### PR TITLE
Fix "no replicas available" sql hint to be conditional based on managed or unmanaged cluster

### PR DESCRIPTION
**Pull request by chaas**
Fix "no cluster replicas available" adapter sql hint to be conditional based on managed or unmanaged cluster, as reported in https://github.com/MaterializeInc/materialize/issues/20000#issuecomment-2197198856.

### Motivation

  * This PR fixes a recognized bug.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: src/adapter/src/error.rs [hunk 1]
```diff
@@ -98,7 +98,7 @@ pub enum AdapterError {
     /// Transaction cluster was dropped in the middle of a transaction.
     ConcurrentClusterDrop,
     /// Target cluster has no replicas to service query.
-    NoClusterReplicasAvailable(String),
+    NoClusterReplicasAvailable(String, bool),
```
Comment (maddyblue) [inline src/adapter/src/error.rs:101]:
```suggestion
    NoClusterReplicasAvailable{
        name: String,
        is_managed: bool,
    },
```

**Review by maddyblue — APPROVED** [review]

File: src/adapter/src/error.rs (hunk 1 above)
Comment (benesch) [reply src/adapter/src/error.rs:101]: Strong +1 on this.

**Review by benesch — APPROVED** [review]
This is awesome stuff, @chaas! Thanks for just knocking this one out.

File: src/adapter/src/error.rs [hunk 2]
```diff
@@ -367,8 +367,13 @@ impl AdapterError {
...
                 expected.join(", ")
             )),
-            AdapterError::NoClusterReplicasAvailable(_) => {
-                Some("You can create cluster replicas using CREATE CLUSTER REPLICA".into())
+            AdapterError::NoClusterReplicasAvailable(_, is_managed) => {
+                Some(if *is_managed {
+                    "You can adjust the replication factor of the cluster with ALTER CLUSTER. \
+                    For example `ALTER CLUSTER <cluster-name> SET (REPLICATION FACTOR 1)`".into()
+                } else {
+                    "You can create cluster replicas using CREATE CLUSTER REPLICA".into()
```
Comment (morsapaes) [inline src/adapter/src/error.rs:375]:
```suggestion
                    "Use ALTER CLUSTER to adjust the replication factor of the cluster. \
                    Example:`ALTER CLUSTER <cluster-name> SET (REPLICATION FACTOR 1)`".into()
                } else {
                    "Use CREATE CLUSTER REPLICA to attach clusters replicas to the cluster ".into()
```

Comment (morsapaes) [reply src/adapter/src/error.rs:375]: Feel free to ignore the suggestion if this disrupts the current style of hints, but this feels more direct than the alternative, using fewer words.

File: src/adapter/src/error.rs (hunk 1 above)
Comment (chaas) [reply src/adapter/src/error.rs:101]: Thanks! My Rust is a bit rusty these days 🥁

**Review by maddyblue — APPROVED** [review]

**Pull request merged**